Common/Util/ExceptionParser.cs

```diff
@@ -0,0 +1,181 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Python.Runtime;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Util
+{
+    public class ExceptionParser : Exception
```

Comment:
Let's remove this inheritance here... and `ExceptionParser` is not an `Execption`.

Perhaps `ExceptionParser` can be a stand alone class with a single, state-less method (state-less methods are very easily unit tested):
``` csharp
Exception ParseException(Exception error);
```

The `ExceptionParser` can decide whether or not to invoke python/csharp parsers, or perhaps it doesn't invoke them at all if it's not a parsable exception.